crates/core/src/operations/update.rs

```diff
@@ -231,15 +247,23 @@ async fn execute(
         .await?;
     let scan = Arc::new(scan);
 
-    // Create a projection for a new column with the predicate evaluated
-    let input_schema = snapshot.input_schema()?;
+    // Wrap the scan with a CDCObserver if CDC has been abled so that the tracker can
+    // later be used to produce the CDC files
+    let scan: Arc<dyn ExecutionPlan> = match snapshot.table_config().enable_change_data_feed() {
```

Comment:
If the table is at v7 writer, then we should only look at the writer_feature whether it exists or not.

In spark it's possible to disable change data feed through the config, even though the writer feature exists.

Reply:
I'm confused by this comment. Are you stating that our enable_change_data_feed() function is incorrect?  I understand what portion of the protocol you're referring to here, but I'm not clear what change is being suggested

Reply:
That function only grabs it from the config, which is valid for writer versions 1..6, but for writer version >=7 we need to look at the writer's features in the protocol if ChangeDateFeed is enabled there.